src/rust/ensogl/lib/core/src/system/gpu/data/buffer.rs

```diff
@@ -450,3 +451,31 @@ pub trait IsBuffer {
     fn vertex_attrib_pointer (&self, index:u32, instanced:bool);
     fn set_to_default        (&self, index:usize);
 }
+
+// Calls are not recursive, as inherent methods are preferred over methods provided by trait.
```

Comment:
I dont understand this code. Why something like the default impl of e.g. `update` would ever be useful? Using the default one will cause recursion, right?

Reply:
Write why needed in a comment